uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkPattern.php

```diff
@@ -135,14 +135,23 @@ public static function validateDefaultWorkPattern($params) {
       );
     }
 
-    if($isUpdate && $isToBeDisabled) {
-      $isDefaultWorkPattern = self::isDefault($params['id']);
+    if(!$isUpdate) {
+      return;
+    }
 
-      if ($isDefaultWorkPattern) {
-        throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkPatternException(
-          'You cannot disable the default Work Pattern'
-        );
-      }
+    $isDefaultWorkPattern = self::isDefault($params['id']);
+
+    if($isToBeDisabled && $isDefaultWorkPattern) {
+      throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkPatternException(
+        'You cannot disable the default Work Pattern'
+      );
+    }
+
+    $isToBeUncheckedAsDefault = isset($params['is_default']) && !$params['is_default'];
+    if($isToBeUncheckedAsDefault && $isDefaultWorkPattern) {
+      throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkPatternException(
+        'It is not possible to have no default Work Pattern'
```

Comment:
Does this also implies `Default Work Pattern required.`? Guess it will be more simplified if it means same thing.

Reply:
Yes a default Work pattern is required. A default work pattern is actually shipped with the default CiviHR installation